File: 1-TwoSum/1-TwoSum.py
# Last updated: 9/14/2025, 4:58:15 AM
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:


        n = len(nums)
        for i in range(n):
            for j in range(i+1, n):
                if nums[i] + nums[j] == target:
                    return [i,j]
                    print(i)
        return None
             
        

File: 1-TwoSum/1-TwoSum.java
// Last updated: 5/7/2025, 8:04:27 PM
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> result= new HashMap<>();
        
        for (int i = 0; i<nums.length; i++){
            int diff = target - nums[i];
            if (result.containsKey(diff)){
                return new int [] { result.get(diff), i};
            }
            else{
                result.put(nums[i], i);
            }
        } return new int [] {};
    }
}